presto-main/src/test/java/io/prestosql/execution/buffer/BenchmarkPagesSerde.java

```diff
@@ -0,0 +1,217 @@
+ * limitations under the License.
+ */
+package io.prestosql.execution.buffer;
+
+import com.google.common.collect.ImmutableList;
+import io.prestosql.spi.Page;
+import io.prestosql.spi.PageBuilder;
+import io.prestosql.spi.block.BlockBuilder;
+import io.prestosql.spi.type.RowType;
+import io.prestosql.spi.type.Type;
+import io.prestosql.spiller.AesSpillCipher;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.infra.Blackhole;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import org.openjdk.jmh.runner.options.VerboseMode;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.Random;
+
+import static io.airlift.slice.Slices.utf8Slice;
+import static io.prestosql.metadata.MetadataManager.createTestMetadataManager;
+import static io.prestosql.spi.type.BigintType.BIGINT;
+import static io.prestosql.spi.type.VarcharType.VARCHAR;
+import static java.nio.charset.StandardCharsets.ISO_8859_1;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+@State(Scope.Thread)
+@OutputTimeUnit(SECONDS)
+@Fork(1)
+@Warmup(iterations = 10, time = 1, timeUnit = SECONDS)
+@Measurement(iterations = 10, time = 1, timeUnit = SECONDS)
+@BenchmarkMode(Mode.Throughput)
+public class BenchmarkPagesSerde
+{
+    @Benchmark
+    public void serialize(BenchmarkData data, Blackhole bh)
+    {
+        Page[] pages = data.dataPages;
+        PagesSerde serde = data.serde;
+        for (Page p : pages) {
+            bh.consume(serde.serialize(p));
+        }
+    }
+
+    @Benchmark
+    public void deserialize(BenchmarkData data, Blackhole bh)
+    {
+        SerializedPage[] pages = data.serializedPages;
+        PagesSerde serde = data.serde;
+        for (SerializedPage p : pages) {
+            bh.consume(serde.deserialize(p));
+        }
+    }
+
+    @State(Scope.Thread)
+    public static class BenchmarkData
+    {
+        private static final int ROW_COUNT = 10000;
+        private static final RowType rowType = RowType.anonymous(ImmutableList.of(VARCHAR, VARCHAR, VARCHAR, VARCHAR));
+        private static final List<Type> TYPES = ImmutableList.of(BIGINT, rowType, rowType, rowType);
+        @Param({"true", "false"})
+        private boolean encrypted;
+        @Param({"true", "false"})
+        private boolean compressed;
+        @Param("1000")
```

Comment:
Just make it a field as in `BenchmarkData`, see: `io.prestosql.orc.BenchmarkColumnReaders.BenchmarkData#random`